redox/processor_scheduled.go

```diff
@@ -117,3 +125,20 @@ func (r *scheduledSummaryAndReportProcessor) getClinicSettings(ctx context.Conte
 
 	return resp.JSON200, nil
 }
+
+func patientHasUploadedDataRecently(patient clinics.Patient) bool {
+	cutoffDate := time.Now().Add(-recentDataCutoff)
+	mostRecentUploadDate := getMostRecentUploadDate(patient)
+	return mostRecentUploadDate.After(cutoffDate)
+}
+
+func getMostRecentUploadDate(patient clinics.Patient) time.Time {
+	var mostRecentUpload time.Time
+	if patient.Summary != nil && patient.Summary.CgmStats != nil && patient.Summary.BgmStats.Dates.LastUploadDate != nil {
```

Comment:
```suggestion
	if patient.Summary != nil && patient.Summary.CgmStats != nil && patient.Summary.CgmStats.Dates.LastUploadDate != nil {
```

Reply:
Was this a typo, or intentional?

Reply:
You could add a test with different dates for each, then check the value that's returned. Probably not a super important test, but, assuming the above is a typo, that would have caught it.